scipy/stats/_qmc.py

```diff
@@ -898,6 +988,151 @@ def random(self, n: IntNumber = 1) -> np.ndarray:
         return samples  # type: ignore[return-value]
 
 
+class OptimalLatinHypercube(QMCEngine):
+    """Greedy optimization of a Latin hypercube sample (OLHS).
+
+    Optimize the sample by doing random permutations of coordinates to lower
+    the centered discrepancy [1]_. The resulting sample is still a LHS.
```

Comment:
Yes the equation is in there. It is true that the article refers to the fancy algorithm. I can put a more general one if you want. Although I am kind of hopping that interested people to actually go read this would find a way to do it right?? 😅

Reply:
K, thought it might have been left over from before. Yes, the algorithm is not very complicated : )